### PR TITLE
feature/add-get-stake-reward: add GET_STAKE_REWARD and test for it

### PR DESCRIPTION
**Pull request by Verbalist**

### CODE REVIEW

File: docker-entrypoint-new.sh [hunk 1]
```diff
@@ -14,7 +14,9 @@ else
     fi
     sleep 5
     if [ "$MODE" == "TEST" ]; then
-        npm run test
+        wait-port "$API_HOST:${API_PORT:-7008}"
+        wait-port "$CORE_HOST:${CORE_RPC_PORT:-7009}"
+        npm run test:api
```
Comment (macaqueee) [inline docker-entrypoint-new.sh]: maybe we need to create some environments like `TEST_API`?

File: test/api/base/mock.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { Timestamp } from 'shared/model/account';
+import config from 'shared/config';
+
+const TWO = 2;
+const FIVE = 5;
+const TEN = 10;
+const S = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+
+export const randStr = (range) =>
+    Array(range)
```
Comment (macaqueee) [inline test/api/base/mock.ts]: formatter?

File: test/api/base/fixture.ts [hunk 3]
```diff
@@ -9,4 +20,58 @@ export class Fixture {
             type: MessageType.REQUEST
         };
     }
+
+    static async createBlock() {
+        const rawBlock: any = await db.one('INSERT INTO block' +
```
Comment (macaqueee) [inline test/api/base/fixture.ts]: extract to `query`

File: test/api/base/mock.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { SECOND } from 'core/util/const';
```
Comment (macaqueee) [inline test/api/base/mock.ts]: rename to `util`

File: api/repository/transaction/index.ts [hunk 5]
```diff
@@ -46,6 +46,30 @@ class TransactionPGRepository {
         };
 
     }
+
+    async getVotesWithStakeReward(senderPublicKey: string, limit: number, offset: number):
+        Promise<{ transactions: Array<Transaction<IAssetVote>>, count: number }> {
+        const transactions = await db.manyOrNone(query.getVoteWithStake, {
```
Comment (axelsheva) [inline api/repository/transaction/index.ts]: getVote`s`WithStake

**Pull request merged**